pymtl3/stdlib/ifcs/CalleeIfc.py

```diff
@@ -0,0 +1,167 @@
+#=========================================================================
+# CalleeIfc.py
+#=========================================================================
+# RTL method interface. RTL equivalence of NonBlockingCalleeIfc in CL
+#
+# Author : Yixiao Zhang
+#   Date : June 10, 2019
+
+from __future__ import absolute_import, division, print_function
+
+from copy import deepcopy
+
+from pymtl3 import *
+from pymtl3.dsl.errors import InvalidConnectionError
+
+
+#-------------------------------------------------------------------------
+# CalleeRTL2CL
+#-------------------------------------------------------------------------
+class CalleeRTL2CL( Component ):
+
+  def construct( s, ifc_rtl_callee ):
```

Comment:
I think for adapter we usually take types as arguments. See here. If you pass in ArgsType and RetType you don't need deepcopy and inverse

Reply:
When you instantiate an adapter (e.g. https://github.com/cornell-brg/pymtl3/blob/master/pymtl3/stdlib/ifcs/SendRecvIfc.py#L49), you actually have the MsgType from when the interface is instantiated (https://github.com/cornell-brg/pymtl3/blob/master/pymtl3/stdlib/ifcs/SendRecvIfc.py#L30). Thus you can pass the interface Type into the adapter. Sometimes we even pass in two types https://github.com/cornell-brg/pymtl3/blob/master/pymtl3/stdlib/ifcs/mem_ifcs.py#L44.